### PR TITLE
Minor Coverity Scan fixes

### PR DESCRIPTION
**Pull request by troglobit**
## Description

Fixes minor Coverity weekly scan issues as well as some linter warnings from clang-tidy.

## Checklist

Tick *relevant* boxes, this PR is-a or has-a:

- [X] Bugfix
  - [ ] Regression tests
  - [ ] ChangeLog updates (for next release)
- [ ] Feature
  - [ ] YANG model change => revision updated?
  - [ ] Regression tests added?
  - [ ] ChangeLog updates (for next release)
  - [ ] Documentation added?
- [X] Code style update (formatting, renaming)
- [ ] Refactoring (please detail in commit messages)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: src/confd/src/ietf-syslog.c [hunk 1]
```diff
@@ -109,8 +115,10 @@ static size_t selector(sr_session_ctx_t *session, struct action *act)
 
 		fprintf(act->fp, "%s%s.%s", i ? ";" : "", fxlate(facility), sxlate(severity));
 
-		free(facility);
-		free(severity);
+		if (facility)
```
Comment (mattiaswal) [inline src/confd/src/ietf-syslog.c:118]: Put this check after this line instead:
facility = srx_get_str(session, "%s/facility", entry->xpath);
 why continue to write config after that if null?

File: src/confd/src/ietf-syslog.c [hunk 2]
```diff
@@ -109,8 +115,10 @@ static size_t selector(sr_session_ctx_t *session, struct action *act)
...
+			free(facility);
+		if (severity)
+			free(severity);
```
Comment (mattiaswal) [inline src/confd/src/ietf-syslog.c:121]: Same as above

File: src/confd/src/ietf-syslog.c (hunk 1 above)
Comment (troglobit) [reply src/confd/src/ietf-syslog.c:118]: OK

**Review by mattiaswal — APPROVED** [review]
:1st_place_medal:

**Pull request merged**